src/Compilers/CSharp/Portable/Compilation/MethodBodySemanticModel.cs

```diff
@@ -9,6 +9,26 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed class MethodBodySemanticModel : MemberSemanticModel
     {
+#nullable enable
+        internal readonly struct InitialState
+        {
+            internal readonly CSharpSyntaxNode Syntax;
+            internal readonly BoundNode? Body;
+            internal readonly ExecutableCodeBinder? Binder;
+            internal readonly NullableWalker.SnapshotManager? SnapshotManager;
+            internal readonly ImmutableDictionary<Symbol, TypeWithAnnotations>? AnalyzedVariableDeclarations;
+
+            internal InitialState(CSharpSyntaxNode syntax, BoundNode? bodyOpt = null, ExecutableCodeBinder? binder = null, NullableWalker.SnapshotManager? snapshotManager = null, ImmutableDictionary<Symbol, TypeWithAnnotations>? analyzedVariableDeclarations = null)
```

Comment:
>NullableWalker.SnapshotManager? snapshotManager = null, ImmutableDictionary<Symbol, TypeWithAnnotations>? analyzedVariableDeclarations = null [](start = 123, length = 141)

Are these arguments used?

Reply:
Yes, they're used in MethodCompiler.

---
In reply to: [305047427](https://github.com/dotnet/roslyn/pull/37322#discussion_r305047427) [](ancestors = 305047427)